src/wheel_integration_package/src/IntegrationNode.cpp

```diff
@@ -0,0 +1,79 @@
+/*
+ * Created By: Vijeeth Vijhaipranith
+ * Created On: Oct 30th, 2020
+ * Description: An example node that subscribes to a topic publishing strings,
+ *              and re-publishes everything it receives to another topic with
+ *              a "!" at the end
+ */
+
+#include <IntegrationNode.h>
+
+
+MyClass::MyClass(int argc, char **argv, std::string node_name, float dist, float max_speed) {
+    // Setup NodeHandles
+    ros::init(argc, argv, node_name);
+    ros::NodeHandle nh;
+    ros::NodeHandle private_nh("~");
+
+    distBetweenWheels = dist;
+    maximum_speed = max_speed;
+
+    // Setup Subscriber(s)
+    std::string topic_to_subscribe_to = "cmd_vel";
+    int queue_size = 10;
+    my_subscriber = nh.subscribe(topic_to_subscribe_to, queue_size, &MyClass::subscriberCallBack, this);
+
+    // Setup Publisher(s)
+    std::string lwheels_topic = private_nh.resolveName("lwheels_pub_topic");
+    int lwheels_queue_size = 1;
+    lwheels_publisher = private_nh.advertise<geometry_msgs::Twist>(lwheels_topic, lwheels_queue_size);
+
+    std::string rwheels_topic = private_nh.resolveName("rwheels_pub_topic");
+    int rwheels_queue_size = 1;
+    rwheels_publisher = private_nh.advertise<geometry_msgs::Twist>(rwheels_topic, rwheels_queue_size);
+}
+
+/*
+ * Requires:
+ *      -1.0 <= linear.x 1.0
+ *      -Z_SENSITIVITY <= angular.x Z_SENSITIVITY
+ * Returns:
+ *      Publishes the linear velocity of left and right wheel in a Twist message
+ */
+void MyClass::subscriberCallBack(const geometry_msgs::Twist::ConstPtr& msg) {
```

Comment:
This function and the corresponding tests are well crafted, but because of the lack of commenting, it feels like magic. Ideally we want the code to be easily maintained and not require too much thought to fully understand. Please comment here and in the test file so that it's clear what's going on to somebody who wasn't present while you were creating this.